### ISSUE TITLE
Crash on Android 6 Api 23

### ISSUE THREAD

**Issue by KinjiQaudra** [mauron85/cordova-plugin-background-geolocation#21]
When using the ionic application in Android 6 API 23, cordova-plugin-background-geolocation crash and applicaiton stop, have you a solution for this issue, thanks

**Comment by mauron85** [comment]
Can you send me output from adb logcat?

**Comment by KinjiQaudra** [comment]
E/SysUtils: ApplicationContext is null in ApplicationStatus
E/chromium: [ERROR:browser_gpu_channel_host_factory.cc(258)] Failed to init browser shader disk cache.
E/libEGL: validate_display:255 error 3008 (EGL_BAD_DISPLAY)
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.
E/cr.LocationProvider: Caught security exception while registering for location updates from the system. The application does not have sufficient geolocation permissions.

**Comment by KinjiQaudra** [comment]
I think is the new système of permission android 6 http://developer.android.com/about/versions/marshmallow/android-6.0-changes.html

**Comment by mauron85** [comment]
Hi, I've just succefully tested SampleApp on android 6 emulator. Can you try to install SampleApp and check if it is woking?

https://github.com/mauron85/cordova-plugin-background-geolocation/tree/master/example/SampleApp

**Comment by KinjiQaudra** [comment]
I have the same error with exemple, application stop after 10 sec !

**Comment by KinjiQaudra** [comment]
I have a problem when start the background traking

**Comment by KinjiQaudra** [comment]
````
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager: Uncaught exception from plugin
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager: java.lang.IllegalArgumentException: Receiver not registered: com.tenforwardconsulting.cordova.bgloc.BackgroundGeolocationPlugin$1@382b7dc
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:780)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1195)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:576)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at com.tenforwardconsulting.cordova.bgloc.BackgroundGeolocationPlugin.execute(BackgroundGeolocationPlugin.java:142)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:95)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.apache.cordova.PluginManager.exec(PluginManager.java:130)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:37)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.os.Handler.dispatchMessage(Handler.java:102)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.os.Looper.loop(Looper.java:148)
12-18 16:25:05.676 11238-11296/com.transistorsoft.backgroundgeolocation E/PluginManager:     at android.os.HandlerThread.run(HandlerThread.java:61)
```
The log cat is different with exemple, the application dont ask to the user to accepte geolocalisation like applications in android M

**Comment by mauron85** [comment]
Oh, You have been hit by new bug, I've unintentionally introduces in version 0.8.2. It has nothing todo with Android 6. I'm about fix it, I'll let you know so you can run SampleApp again.

In mean time, please send me your device manufucaturer, model and android version ...

Also I believe this plugin is not asking for any permissions. Are you sure, you're not using navigator.geolocation.getCurrentPosition, watchPosition ....?

**Comment by KinjiQaudra** [comment]
Nexus 5x, Android 6.0.1, i use navigator.geolocation.getCurrentPosition, but when i comment bg.start(), the problème disappear, i use android target 23 because i have plugin require api 23, sorry for my english :)

**Comment by mauron85** [comment]
Version 0.8.3. Can you try SampleApp. Need to remove plugin and readd to get new version.

```
cordova plugin rm cordova-plugin-mauron85-background-geolocation
cordova plugin add cordova-plugin-mauron85-background-geolocation@0.8.3
```

**Comment by christocracy** [comment]
I've have had this fixed since November 29th in my [PremiumVersion](http://shop.transistorsoft.com/pages/cordova-background-geolocation-premium), building for **API 23**

See [here](http://cordova.apache.org/docs/en/latest/guide/platforms/android/plugin.html).  Search "Android Permissions"

**Comment by mauron85** [comment]
fixed in 0.9.4. Reopen if you find any more issues.

**Comment by KinjiQaudra** [comment]
It work, thanks ;)
